src/App.tsx

```diff
@@ -15,47 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_ALPHABET = 'alphabet',
+  SORT_LENGTH = 'length',
+  SORT_DEFAULT = '',
+}
+
+function getvisibleGoods(
+  goods: string[],
+  sortField: SortType,
+  toReverse: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortField) {
+    visibleGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (toReverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.SORT_DEFAULT);
+  const [isReversed, setReversedField] = useState(false);
+
+  const goods = getvisibleGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+
+  const handleResetClick = () => {
+    setSortField(SortType.SORT_DEFAULT);
+    setReversedField(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortType.SORT_ALPHABET })}
+          onClick={() => setSortField(SortType.SORT_ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SortType.SORT_LENGTH })}
+          onClick={() => setSortField(SortType.SORT_LENGTH)}
```

Comment:
it's a good practice to move click handlers into own functions 
Example: 
```
const handleSortClick = (sortType) => () => {
  setSortField(sortType)
}
```

```suggestion
          onClick={handleSortClick(SortType.SORT_LENGTH)}
```